### PR TITLE
gocyclo: 2015-02-08 -> 0.4.0

### PR DESCRIPTION
**Pull request by vs49688**
###### Motivation for this change

Updating.

We're pulling from a random fork, so switch to the actual upstream.

Also change to use buildGoModule instead of buildGoPackage.

Should supersede https://github.com/NixOS/nixpkgs/pull/145102.

@stackshadow I can add `Co-Authored-By:` if you'd like.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by stackshadow** [comment]
Would be cool, i didn't had the time yet to maintain my last PR for this package. Was a very busy month 😄

File: pkgs/development/tools/gocyclo/default.nix [hunk 1]
```diff
@@ -1,23 +1,20 @@
...
-    sha256 = "094rj97q38j53lmn2scshrg8kws8c542yq5apih1ahm9wdkv8pxr";
+    rev = "v${version}";
+    sha256 = "1s9m5m5p76wcxi5n4diz891kd5db4ll21fsh9fnvvf9w7yrmgdw2";
   };
 
+  vendorSha256 = "0sjjj9z1dhilhpc8pq4154czrb79z9cm044jvn75kxcjv6v5l2m5";
+
   meta = with lib; {
     description = "Calculate cyclomatic complexities of functions in Go source code";
     homepage = "https://github.com/alecthomas/gocyclo";
```
Comment (bobby285271) [inline pkgs/development/tools/gocyclo/default.nix:20]:
```suggestion
    homepage = "https://github.com/fzipp/gocyclo";
```

Comment (vs49688) [reply pkgs/development/tools/gocyclo/default.nix:20]: Oh fiddlesticks, I fixed that originally, but must've force-pushed over it...

Comment (vs49688) [reply pkgs/development/tools/gocyclo/default.nix]: Fixed

**Comment by bobby285271** [comment]
@ofborg build gocyclo

**Comment by stackshadow** [comment]
grafik

Okay ther is nothing changed according to the "old" cli-interface.

But the calulation is different 🤔 
grafik

**Pull request merged**